Adding new features

hannahc243 opened the pull request:
This code builds on the previous code base by adding more information to the final flare list output including the flare locations in the STIX reference frame, the locations from Earth in helioprojective coordinates, flare locations in heliographic coordinates, the light time correction and peak counts in non thermal energy channels etc.

hannahc243 committed:
adjusting to start of January 2021

hannahc243 committed:
Merge branch 'main' of https://github.com/hayesla/stix_flarelist_science into adding_earth

hannahc243 committed:
added new features in final flare list and did analysis for 09-12 nov 22 events

hannahc243 committed:
adding location in heliographic stonyhurst

hannahc243 committed:
work jan 2021 - feb 2023

hannahc243 committed:
added locations from earth, in stix frame and light time corrections

hannahc243 committed:
remove stix api query file from remote repo

hannahc243 committed:
pushed wrong files, rm from remote

hannahc243 committed:
pushed wrong files, rm from remote

hannahc243 committed:
pushed wrong files, rm from remote

hannahc243 committed:
final flare list

hannahc243 committed:
pushed wrong files, rm from remote

hannahc243 committed:
pushed wrong files, rm from remote

hannahc243 committed:
check_locations script

hannahc243 committed:
location check code

hayesla commented:
thanks for this @hannahc243 !

Could you do me a favour and give a description of whats been updated? and what the individual files/notebooks do and details of the csv files? it will just help with review :)

hannahc243 commented:
Yes, of course!

1. In **get_fullflarelist_step1.py** the only thing that has been modified is the start and end times to include a wider time range. 
2. In **get_available_datalist_stix_step2.py**, the csv file read in has been changed to the output from step 1 and I adjusted the counts threshold to >5e2 peak counts in the 4-10 keV bin. Similarly for **get_sciencedatatable_datacenter_step1_5.py**.
3.  In **download_files_from_list.ipynb**: the path to csv was changed. I added the three more non thermal channel quicklook peak counts to the dataframe. 
4. **loop_files_2.pro** the field numbers have been adjusted because of the new columns in the csv output from the previous step. 
5. **stx_estimate_flare_location_flare_list.pro** has been adapted to return the flare location in the stix reference frame (stix_max_bp_coord). 
6. **get_flare_location.pro** now calls **stx_get_header_corrections.pro** in order to extract the light time correction. Now two csv files are written (_flare_location_output4_part1.csv_ and _flare_location_output4_part2.csv_), the first with the previous entries and the second with the flare id for cross checking and the new attributes: the flare locations in the stix reference frame (max_bp_coord_stix) and the light time correction (time shift). Ideally this would all be in one csv but it seems there is a limit of 8 columns in a csv file write with the write_csv function. So I wrote two separate ones and combined in the next step as a work around. We can figure out a better way to do this!
7. The next notebook (**combine_flare_lists.ipynb**) is a new notebook for combining the two csv files and adding this information to the dataframe output in step 3. This code also finds the flare location in Earth based helioprojective coordinates and heliographic stonyhurst coordinates. Finally, the flare locations in the new coordinate frames are appended to the dataframe which is then saved out as the final flare list for analysis.

Hope this helps! Let me know if there are outstanding questions.
